pkg/ddl/ddl.go

```diff
@@ -831,6 +831,7 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
 		if err != nil {
 			logutil.BgLogger().Error("error when getting the ddl history count", zap.Error(err))
 		}
+		d.runningJobs.clear()
```

Comment:
Now we reuse the `startDispatchLoop` to initiate the `runningJobs`, by relying on the deterministic order from `order by processing desc, job_id` and add the jobs one by one. 

Considering the case that the former DDL owner marked wrong jobs as running, like
100 (running), 101 (pending), 102 (running wrongly). The correct state should be (running, pending, pending) then (finished, running, pending). I slightly prefer the new DDL owner re-compute the running jobs instead of reuse the persistent state from persistent table.

Reply:
“Considering the case that the former DDL owner marked wrong jobs as running, like
100 (running), 101 (pending), 102 (running wrongly).”
How come？

Reply:
> “Considering the case that the former DDL owner marked wrong jobs as running, like 100 (running), 101 (pending), 102 (running wrongly).” How come？

The old owner uses the code before this PR, and marked 102 as running